### PR TITLE
Refresh Token

### PR DESCRIPTION
**Pull request by 1mr124**
This is the **First Prototype** of the Refresh Token Logic (BackEnd Based).

**Next:**

- Ensure that all exceptions are handled correctly.
- Add the Refresh Token to the API (authApi).

**Side Note:** Asking for your overall opinion so far.

### CODE REVIEW

File: FlaskSite/api/v1/auth.py [hunk 1]
```diff
@@ -27,3 +27,16 @@ def login():
...
+
+
+@bp.route('/auth/refresh', methods=['POST'])
+def refresh_token():
+    # validate if the cookie has the refresh token
+    # to-do is to move refresh_token to constants.py
+    refresh_token = request.cookies.get('refresh_token')
+
+    if not refresh_token:
+        return jsonify({"msg": "Missing refresh token"}), 401
```
Comment (MohamedSamehMohamed) [inline FlaskSite/api/v1/auth.py]: "message"

File: FlaskSite/controllers/auth_controller.py [hunk 2]
```diff
@@ -15,13 +15,48 @@ def register(username, password, email):
...
+                httponly=True,
+                secure=secure_cookie,  # Enforces HTTPS in production
+                samesite='None' if secure_cookie else 'Lax',  # Use 'None' for cross-site requests in production
+                path='/'
+            )
+            
+        response.set_cookie('refresh_token', refresh_token)
+        return response, 200
     except Exception as e:
         return jsonify({"message": f"An error occurred {e}"}), 500
```
Comment (MohamedSamehMohamed) [inline FlaskSite/controllers/auth_controller.py]: don't return exception in response

File: FlaskSite/controllers/auth_controller.py [hunk 3]
```diff
@@ -15,13 +15,48 @@ def register(username, password, email):
...
+        success, result = auth_service.refresh_access_token(refresh_token)
+        if not success:
+            if result == "Token expired":
+                # Clear the expired refresh token in the response
+                response = jsonify({"message": "Refresh token expired"})
+                response.status_code = 401
+                response.delete_cookie('refresh_token')  # Optional: clear cookie if used
+                return response
+            else:
+                return jsonify({"message": result}), 401  # Other token-related errors
```
Comment (MohamedSamehMohamed) [inline FlaskSite/controllers/auth_controller.py:58]: either modify refresh_access_token to not return str(e) in result or don't put exception in message

File: FlaskSite/controllers/auth_controller.py [hunk 4]
```diff
@@ -15,13 +15,48 @@ def register(username, password, email):
...
     except Exception as e:
         return jsonify({"message": f"An error occurred {e}"}), 500
+
+
+
+def refresh_access_token(refresh_token):
+    try:
+        success, result = auth_service.refresh_access_token(refresh_token)
+        if not success:
+            if result == "Token expired":
```
Comment (MohamedSamehMohamed) [inline FlaskSite/controllers/auth_controller.py]: make result enum or remove magic string

File: client/src/ApiTest/testApi.mjs [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+  refreshAccessToken: async () => {
+    try {
+      // Make a request to refresh the access token
+      const response = await publicApi.post(
+        '/auth/refresh',
+        {},
+        {
+          headers: {
+            Cookie: 'refresh_token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJmcmVzaCI6ZmFsc2UsImlhdCI6MTczMDU1NDI0MCwianRpIjoiNWExM2FiMDUtMjM1Ny00NDFiLWFhODgtMzAzODA0NjFlYzQzIiwidHlwZSI6InJlZnJlc2giLCJzdWIiOjEsIm5iZiI6MTczMDU1NDI0MCwiY3NyZiI6IjI4MDUwOTdhLWE5MzItNGQxNi05OTc0LTU5MjUyZDE3OTQ2NCIsImV4cCI6MTczMTE1OTA0MH0.VoaaWYFwi7gAVrFKyvPu7CvQscmppESubVV5Ef07YpM'
```
Comment (MohamedSamehMohamed) [inline client/src/ApiTest/testApi.mjs]: what is this?

File: FlaskSite/controllers/auth_controller.py (hunk 4 above)
Comment (1mr124) [reply FlaskSite/controllers/auth_controller.py]: i will resolve all the magic strings later in one branch

File: client/src/ApiTest/testApi.mjs (hunk 5 above)
Comment (1mr124) [reply client/src/ApiTest/testApi.mjs]: For testing purposes only, I want it to be accessible on all devices.

Comment (MohamedSamehMohamed) [reply client/src/ApiTest/testApi.mjs]: is it a valid token? won't it expires?

File: FlaskSite/controllers/auth_controller.py (hunk 4 above)
Comment (MohamedSamehMohamed) [reply FlaskSite/controllers/auth_controller.py]: resolve the old ones, but don't add new magic strings

File: client/src/ApiTest/testApi.mjs (hunk 5 above)
Comment (1mr124) [reply client/src/ApiTest/testApi.mjs]: Just like curl.txt examples

File: FlaskSite/controllers/auth_controller.py (hunk 4 above)
Comment (1mr124) [reply FlaskSite/controllers/auth_controller.py]: A Good Point, but...

**Review by MohamedSamehMohamed — APPROVED** [review]

**Pull request merged**